api/src/main/java/org/openmrs/module/oauth2login/PropertyUtils.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.oauth2login;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.openmrs.util.OpenmrsUtil;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Properties;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class PropertyUtils {
+	
+	protected static final Log LOG = LogFactory.getLog(PropertyUtils.class);
+	
+	private static final Pattern ENV_PATTERN = Pattern.compile("\\$\\{([^}]+)}");
+	
+	public static Properties getProperties(Path path) throws IOException {
```

Comment:
Did the ticket say you need to support env variables in the auth2.properties? Or does it say support disabling via env variable? Because then all you need to do is check if the env variable has been set or not, that way you don't have to deal with all this extra env substitution logic.

Reply:
If you closely examine the changes, you'll notice this isn't a new introduction. I've refactored the logic into a new class, `PropertyUtils`, to enable reusing it. The ability to resolve environment variables was introduced in this [commit](https://github.com/openmrs/openmrs-module-oauth2login/commit/62ade3cf1898aab6e716564c037a2a7c6dcfb8b1).